{{ cookiecutter.dir_name }}/AndroidManifest.xml

```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="{{ cookiecutter.bundle }}.{{ cookiecutter.app_name }}">
+    <uses-permission android:name="android.permission.INTERNET" />
```

Comment:
I'm hesitant to encode this as a hard permission requirement. Not all apps need INTERNET permission; and if we start hard coding permissions, every app is going to get this by default, which is how you get the situation where address book apps ask for access to your camera.

This may require adding a metadata item to cookiecutter (and then to briefcase) where we list permissions that the app requires, so that different projects templates can expose that permission as required.

Reply:
Any ideas on how that metadata thing should work?
My thinking was to add a `template_extra_args` option to briefcase, which would allow sending options down to the template, and then here in the template we would add a parameter `'use_requests_volley'` defaulting to `False` which would condition these changes.
What do you think?

Reply:
In the case of permissions, I think the template needs a high level "permissions" key, whose values are abstracted versions of the permissions offered by iOS, Android etc. Briefcase can then pass in those same permission keys.

I'm not sure how to handle Volley. It's a library that has very specific utility; in the case of a pure Toga app, for example, it might be required as part of the support library, but won't need to exist in *this* form.

Reply:
Hm, I see.
Yeah, it seems tricky.

Since this is currently the way which I know that works for making HTTP requests from an Android app with VOC-compiled code, it seemed to me that it would be not too bad to use this here and use wrappers in the Python side, at least until we have a better way of doing things.

Which form are you imagining for the support library for Toga?

Reply:
If it's in Toga, then it can be implemented in Python - either as part of Toga, or (more likely) as part of the Android support library. The API for Python would be "import requests" or "import httplib". Volley is an implementation detail; the "requires internet access" is an app-level permission (one that every iOS app gets by default, but android apps need to specifically request)

Reply:
I see, thanks!

Yeah, it will be nice when we can do requests with a Pythonic API.

Okay, say that it's a part of the Android support library for Toga which builds on top of Volley (which seems to make sense, since it solves the problem of non-blocking request dispatch with the requests queue thing): wouldn't we still want to add the dependency in here?